### PR TITLE
Pull request to make OpenUCX compile with musl libc and GCC 6.3.0 on 1.3.0-rc3

### PR DESCRIPTION
**Pull request by raphaelcohn**
I'm not a contributor, so, under the terms of the contributor
agreement, I hereby give permission for any other contributor to
submit my changes to OpenUCX as if they were their own.

I'm trying to file a contributor agreement as well.

### CODE REVIEW

**Comment by raphaelcohn** [comment]
This pull request replaces a prior one.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: src/ucs/type/spinlock.h [hunk 1]
```diff
@@ -32,7 +32,7 @@ static inline ucs_status_t ucs_spinlock_init(ucs_spinlock_t *lock)
     }
 
     lock->count = 0;
-    lock->owner = 0xfffffffful;
+    lock->owner = (pthread_t)0xfffffffful;
```
Comment (hoopoepg) [inline src/ucs/type/spinlock.h]: why not PTHREAD_T_NULL?

File: src/ucs/type/spinlock.h [hunk 2]
```diff
@@ -89,7 +89,7 @@ static inline void ucs_spin_unlock(ucs_spinlock_t *lock)
 {
     --lock->count;
     if (lock->count == 0) {
-        lock->owner = 0xfffffffful;
+        lock->owner = (pthread_t)0xfffffffful;
```
Comment (hoopoepg) [inline src/ucs/type/spinlock.h]: the same, may be use PTHREAD_T_NULL instead of -1?

File: src/ucs/type/spinlock.h (hunk 1 above)
Comment (raphaelcohn) [reply src/ucs/type/spinlock.h]: Yep.

File: src/ucs/type/spinlock.h (hunk 2 above)
Comment (raphaelcohn) [reply src/ucs/type/spinlock.h]: Yep.

**Comment by miked-mellanox** [comment]
ok to test

**Comment by shamisp** [comment]
Copyright is missing. Also, did you sign the personal (or corporate) contributor agreement ? thanks !

**Comment by shamisp** [comment]
Ah, missed the note. It will be really helpful to have the contributor agreement. You may email the copy directly to me.

**Comment by raphaelcohn** [comment]
@shamisp - I've sent an email with a copy of the contributor agreement to the address I've got for you in the mailing list - let me know if you don't receive it.

**Comment by raphaelcohn** [comment]
Anything I now need to do?

**Comment by yosefe** [comment]
need to fix the failing tests, for example:
http://bgate.mellanox.com/jenkins/job/hpc-ucx-pr__published_from_mellanox_lab/6609/label=clx-ppc-04.mtl.labs.mlnx,worker=0/console
```
15:12:26 [ RUN      ] signal/test_async.global_timer/0
15:12:26 [clx-ppc-04:9467 :0:9467] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x4)
15:12:27 
15:12:27 /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../src/ucs/async/signal.c: [ ucs_async_signal_sys_timer_delete() ]
15:12:27       ...
15:12:27       170     ucs_trace_func("sys_timer_id=%p", sys_timer_id);
15:12:27       171 
15:12:27       172     ret = timer_delete(sys_timer_id);
15:12:27 ==>   173     if (ret < 0) {
15:12:27       174         ucs_warn("failed to remove the timer: %m");
15:12:27       175     }
15:12:27       176 
15:12:27 
15:12:27 ==== backtrace ====
15:12:27  0 0x000000000003fce4 ucs_async_signal_sys_timer_delete()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../src/ucs/async/signal.c:173
15:12:27  1 0x000000000003d508 ucs_async_add_timer()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../src/ucs/async/async.c:451
15:12:27  2 0x000000001037cf9c base_timer::set_timer()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/ucs/test_async.cc:123
15:12:27  3 0x0000000010372bb0 .LTHUNK15()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/ucs/test_async.cc:172
15:12:27  4 0x00000000100ad468 ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/common/test.cc:270
15:12:27  5 0x000000001037779c test_async::TestBody()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/ucs/test_async.cc:240
15:12:27  6 0x000000001009e634 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/common/gtest-all.cc:3562
15:12:27  7 0x000000001008e0bc testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/common/gtest-all.cc:3635
15:12:27  8 0x000000001008e20c testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/common/gtest-all.cc:3812
15:12:27  9 0x000000001008e458 testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/common/gtest-all.cc:3930
15:12:27 10 0x0000000010094278 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/common/gtest-all.cc:5802
15:12:27 11 0x00000000100946ac testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/common/gtest-all.cc:5719
15:12:27 12 0x0000000010025c04 main()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/clx-ppc-04.mtl.labs.mlnx/worker/0/contrib/../test/gtest/common/gtest.h:20059
15:12:27 13 0x0000000000024980 generic_start_main.isra.0()  libc-start.c:0
15:12:27 14 0x0000000000024b74 __libc_start_main()  ???:0
15:12:27 ===================
15:12:27 Sending notification to raphael.cohn@stormmq.com
```

**Comment by raphaelcohn** [comment]
Hmm... but I didn't change anything to do with `ucs_async_signal_sys_timer_delete`. Or any code to do with PowerPC. This stuff is going to be pretty much impossible for me to sort out.

**Comment by yosefe** [comment]
@raphaelcohn may be related to https://github.com/openucx/ucx/pull/2440/files#diff-956837b588508cb692971bb5bdb8b105R136

**Comment by yosefe** [comment]
@raphaelcohn you replaced timer_create() by a syscall, but timer_delete() still calls glibc

**Comment by shamisp** [comment]
Do we target this PR for v1.3 ?

**Comment by raphaelcohn** [comment]
I'd like it to, but I have to delay working on it for over a week as I'm enjoying looking after my daughters for the hols.

**Comment by shamisp** [comment]
@raphaelcohn it has been hanging here for a while. We just branched for 1.4.x What are the plans for this PR.

**Comment by yosefe** [comment]
closing due to inactivity

**Comment by raphaelcohn** [comment]
When I get time to work on this again, I'll put something together.

**Pull request closed without merging**